### PR TITLE
PMM-3191 Local agent API.

### PR DESCRIPTION
**Pull request by DexterHD**

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

File: api/agentlocalpb/agentlocal.proto [hunk 1]
```diff
@@ -21,13 +21,36 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+import "google/protobuf/timestamp.proto";
+import "inventory/agent_status.proto";
 import "protoc-gen-swagger/options/annotations.proto";
 
-message StatusRequest {}
+// ServerInfo returns information about server which agent was connected.
+message ServerInfo {
+  string url = 1;
+  bool insecure_tls = 2;
+  string last_ping_time = 3;
```
Comment (AlekSi) [inline api/agentlocalpb/agentlocal.proto]: Timestamp

File: api/agentlocalpb/agentlocal.proto [hunk 2]
```diff
@@ -21,13 +21,36 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+  google.protobuf.Timestamp latency = 4;
```
Comment (AlekSi) [inline api/agentlocalpb/agentlocal.proto]: Duration

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

File: api/agentlocalpb/agentlocal.proto [hunk 3]
```diff
@@ -21,13 +21,37 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+// ServerInfo returns information about server which agent was connected.
+message ServerInfo {
+  string url = 1;
+  bool insecure_tls = 2;
+  google.protobuf.Timestamp last_ping_time = 3;
+  google.protobuf.Duration latency = 4;
+}
+
+// AgentInfo returns information about agent which was runned by pmm-agent.
+message AgentInfo {
```
Comment (AlekSi) [inline api/agentlocalpb/agentlocal.proto]: Add agent type

File: api/agentlocalpb/agentlocal.proto [hunk 4]
```diff
@@ -21,13 +21,37 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+  string agent_id = 1;
+  inventory.AgentStatus status = 2;
+  repeated string logs = 3;
+}
+
+// StatusRequest describes request to get local pmm-agent info.
+message StatusRequest {
+  bool show_agent_logs = 1;
```
Comment (AlekSi) [inline api/agentlocalpb/agentlocal.proto]: get_logs

It does not _show_ anything

File: api/agentlocalpb/agentlocal.proto [hunk 5]
```diff
@@ -21,13 +21,37 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
 import "google/api/annotations.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "inventory/agent_status.proto";
 import "protoc-gen-swagger/options/annotations.proto";
 
-message StatusRequest {}
+// ServerInfo returns information about server which agent was connected.
+message ServerInfo {
+  string url = 1;
```
Comment (AlekSi) [inline api/agentlocalpb/agentlocal.proto]: Add version.
Get it from initial server response metadata.

File: api/agentlocalpb/agentlocal.proto (hunk 3 above)
Comment (DexterHD) [reply api/agentlocalpb/agentlocal.proto]: What is agent_type? We didn't talk about it. :/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

File: api/agentlocalpb/agentlocal.proto (hunk 3 above)
Comment (AlekSi) [reply api/agentlocalpb/agentlocal.proto]: You know what agent type is, and it is already known to pmm-agent

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-307/

**Pull request merged**